install/nodejs

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bash
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+# as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+# version.
+
+: ${EBASH_HOME:=$(dirname $0)/..}
+: ${EBASH:=${EBASH_HOME}/share}
+source "${EBASH}/ebash.sh" || { echo "Unable to source ${EBASH}/ebash.sh" ; exit 1 ; }
+
+opt_usage main <<'END'
+This script is used to install Nodejs and npm onto a system. For distros which have a new enough version packaged up in
+the package manager we can simply invoke their installer. Otherwise we install Nodejs from the official upstream
+tarball.
+END
+
+#----------------------------------------------------------------------------------------------------------------------
+#
+# Main
+#
+#----------------------------------------------------------------------------------------------------------------------
+
+default_prefix="/usr/local"
+default_sha_path="/tmp/SHASUMS256.txt"
+nodejs_codename="gallium"
+lts_url="https://nodejs.org/download/release/latest-${nodejs_codename}/"
+
+opt_usage check_for_node_installation <<'END'
+Checks to see if there is already a nodejs installation. If this script was run previously, it will install symlinks
+for the binaries. If this script detects those symlinks, it will remove and re-install based on the latest version
+for the provided lts codename.
+END
+check_for_node_installation()
+{
+    # Grab the paths if they exist
+    nodejs_path=$(which node 2>/dev/null || echo "failed")
+    npm_path=$(which npm 2>/dev/null || echo "failed")
+    nvm_path=$(which nvm 2>/dev/null || echo "failed")
+
+    edebug "$(lval nodejs_path npm_path nvm_path)"
+
+    # If the path check didn't fail then abort (even if a symlink)
+    if [[ "${nvm_path}" != "failed" ]]; then
+        eerror "NVM has been detected as a non-symlink (${nvm_path}), stopping installation."
+        return 1
```

Comment:
Easier testability with these return values.

Reply:
That's a great idea!